geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -209,14 +205,37 @@ class SelectionMapFragment(
         )
     }
 
-    private fun restoreZoomFromPreviousState(state: Bundle) {
+    private fun initializeViewport() {
+        previousState?.let {
+            if (restoreZoomFromPreviousState(it)) {
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized) {
```

Comment:
I added this thinking the intent was not to zoom to the selected feature if there was a viewport to restore. A warning, it would no work with https://github.com/getodk/collect/pull/5095/commits/0277ec3ef3e9a423bf8c32f4a46baa0848512947#diff-41e26ac0b5e19db6790e339c006bc7c1ce451d326733985f3309aa5f3f3d6d89L353 because then the selected feature would not get the bigger marker. Maybe this case should go first? The restore from saved state is on rotation, right?